nfc/libs/NfcCoreLib/lib/Nci/phNciNfc_Init.c

```diff
@@ -738,94 +738,86 @@ phNciNfc_ResetNtfCb(void*     pContext,
         PH_LOG_NCI_INFO_STR("Received RESET notification from NFCC");
 
         if (pNciCtx->dwNtfTimerId != 0)
-        {
             (void)phOsalNfc_Timer_Stop(pNciCtx->dwNtfTimerId);
 
-            if (pTransInfo->wLength >= PHNFCINFC_CORE_RESET_NTF_MIN_LEN)
+        /* Reset Sender statemachine */
+        (void)phNciNfc_CoreResetSenderStateMachine(&pNciCtx->NciCoreContext);
+        (void)phTmlNfc_WriteAbort(pNciCtx->NciCoreContext.pHwRef);
+
+        if (NULL != pNciCtx->tRegListInfo.pResetNtfCb)
+        {
+            pNciCtx->tRegListInfo.pResetNtfCb(pNciCtx->tRegListInfo.ResetNtfCtxt,
+                eNciNfc_NciResetNtf, pInfo, NFCSTATUS_SUCCESS);
+        }
+
+        if (pTransInfo->wLength >= PHNFCINFC_CORE_RESET_NTF_MIN_LEN_2x)
```

Comment:
Shouldn't we check NCI version before comparing pTransInfo->wLength and version-specific constant PHNFCINFC_CORE_RESET_NTF_MIN_LEN_2x ?